### DISCUSSION TITLE
Singularity can't find container

### DISCUSSION THREAD

**Discussion by sara77777777** [nextflow-io/nextflow#4277] [Q&A] [answered]
Hello,
I made a pipeline in which winnowmap is the first process, but when I run the pipeline.nf I get the error:
`winnowmap: command not found`
I belive the problem is that singularity can't find the containers.

This is part of my nextflow.config file: 

```
singularity {
        autoMounts = true
        runOptions = "--bind ${params.bind_path}"
}

process {
        errorStrategy = "ignore"
        withName:MAP { container = "winnowmap:2.03--h43eeafb_2" }      
}


dag { enabled = true }
report { enabled = true }
timeline { enabled = true }
trace { enabled = true }
```

running `echo $SINGULARITY_CACHEDIR` it returns the path to the directory where I have the image: `/work/my_singularity_container`.

This is my image: 
`/work/my_singularity_container/winnowmap-2.03--h43eeafb_2.img`.
I downloaded from here: 
https://depot.galaxyproject.org/singularity

Can someone help me?

**Comment by bentsherman** [comment, marked as answer]
You also need to set `singularity.enabled = true` in your config or use the `-with-singularity` CLI option

**Reply by sara77777777** [reply]
I tried using the `-with-singularity` option but it didn't work, now I'm setting `singularity.enabled = true` in my config and it's working, so thank you for your help

**Reply by colindaven** [reply]
I specify the image directly as well, since I use an NFS storage on a cluster

`process.container = '/mnt/somepath/image.sif'`

**Reply by bentsherman** [reply]
That's a good point. Until recently, containerized runs would fail if you didn't specify a default container with `process.container`. This was fixed in the latest edge release, so you shouldn't need to do that anymore as long as you specify it for each process.
